geode-core/src/main/java/org/apache/geode/cache/query/security/JavaBeanAccessorMethodAuthorizer.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache.query.security;
+
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.Set;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.Declarable;
+
+/**
+ * An immutable and thread-safe {@link MethodInvocationAuthorizer} that allows any method execution
+ * that follows the design patterns for accessor methods described in the JavaBean specification
+ * 1.01, that is, any method whose name begins with "get" or "is". For additional security, only
+ * methods belonging to classes in user-specified packages will be allowed. If a method does not
+ * match the user-specified parameters, then the {@link RestrictedMethodAuthorizer} will make the
+ * final decision
+ *
+ * <p/>
+ *
+ * Some known dangerous methods, like {@link Object#getClass()}, are also rejected by this
+ * authorizer implementation (see
+ * {@link RestrictedMethodAuthorizer#isKnownDangerousMethod(Method, Object)}).
+ * <p/>
+ *
+ * This authorizer implementation addresses only three of the four known security risks:
+ * {@code Java Reflection}, {@code Cache Modification} and {@code Region Modification}.
+ * <p/>
```

Comment:
I'd also add a comment stating that, when used correctly (domain model follows the `JavaBean specification 1.01`), this authorizer address the four security risks. If there are accessors method that can actually modify the object instance, then only 3 out of 4 risks are addressed.

Reply:
Should be fixed in the latest commit